sqlglot/parser.py

```diff
@@ -2734,14 +2734,18 @@ def _parse_table_sample(self, as_modifier: bool = False) -> t.Optional[exp.Table
         )
         method = self._parse_var(tokens=(TokenType.ROW,))
 
-        self._match(TokenType.L_PAREN)
+        matched_l_paren = self._match(TokenType.L_PAREN)
 
         if self.TABLESAMPLE_CSV:
             num = None
             expressions = self._parse_csv(self._parse_primary)
         else:
             expressions = None
-            num = self._parse_primary()
+            num = (
+                self._parse_factor()
+                if self._match(TokenType.NUMBER, advance=False)
+                else self._parse_primary()
```

Comment:
This turned out a bit awkward.. the motivation was to support stuff like `SAMPLE 1 / 10` (shown as an example in ClickHouse's docs). The thing is that `_parse_term` wouldn't suffice because DuckDB also allows `SAMPLE 10%`, so the parser would try to parse this into a `Mod` and fail due to not finding a RHS. We also need to ensure we're calling `_parse_factor` against a number because otherwise it might consume `BUCKET`.